.ci/gitlab/common.yml

```diff
@@ -26,5 +26,9 @@
     - .ci/setup.sh
   after_script:
     - tar -cf - $(ls -d /root/.cabal /root/.stack $(pwd)/.ci/bindist/linux/debian/*/build || true) | zstd -T${THREADS} -3 > cache.tar.zst
+
+.common-local:
```

Comment:
We should probably have some rule of thumb on when to run jobs on public runners / local runners. What about:

```suggestion
# We run tests on local machines if:
# 
#   * A job may take more than 10 minutes to complete on public runners
#   * A job needs specific capabilities public runners cannot provide, e.g.
#     more than 4GB memory of memory, proprietary synthesis tooling.
#
.common-local:
```

Reply:
10 minutes is the timeout. I'd say that if a job is expected to take more than, say, 3 minutes, maybe it's better to do a local one. `prelude:doctests` is already at that limit, btw.

Would we really gain anything by running jobs taking more than that on public runners?

Reply:
Sure, 3, 5, 7 - all works for me :-). As long as there's some guidance.

> Would we really gain anything by running jobs taking more than that on public runners?

My original motivation was moving anything not in the critical time path of a pipeline to public runners. This was before any limits on GitLab (which I'm happy to see implemented, anything to block those damn crypto miners..).